nemo/collections/tts/models/ssl_tts.py

```diff
@@ -0,0 +1,524 @@
+                )
+                content_loader = torch.utils.data.DataLoader(
+                    content_dataset,
+                    batch_size=data_config['batch_size_content'],
+                    collate_fn=content_dataset.general_collate_fn,
+                    shuffle=data_config['shuffle'],
+                    num_workers=data_config.get('num_workers_content', 0),
+                    pin_memory=data_config.get('pin_memory', False),
+                )
+
+        loaders = {"sv": sv_loader, "content": content_loader}
+        return loaders
+
+    def setup_training_data(self, cfg):
+        self._train_dl = self.__setup_dataloader_from_config(self._cfg.train_ds)
+
+    def setup_validation_data(self, cfg):
+        self._validation_dl = CombinedLoader(self.__setup_dataloader_from_config(self._cfg.validation_ds))
+
+    def configure_optimizers(self):
+        optim_backbone_config = self._cfg.optim_backbone.copy()
+        optim_downstream_config = self._cfg.optim_downstream.copy()
+
+        OmegaConf.set_struct(optim_backbone_config, False)
+        sched_backbone_config = optim_backbone_config.pop("sched", None)
+        OmegaConf.set_struct(optim_backbone_config, True)
+
+        OmegaConf.set_struct(optim_downstream_config, False)
+        sched_downstream_config = optim_downstream_config.pop("sched", None)
+        OmegaConf.set_struct(optim_downstream_config, True)
+
+        optim_backbone = instantiate(optim_backbone_config, params=self.encoder.parameters(),)
+        optim_downstream = instantiate(
+            optim_downstream_config,
+            params=itertools.chain(
+                self.downstream_nets.parameters(),
+                self.sv_linear.parameters(),
+                self.content_linear.parameters(),
+                self.sv_loss.parameters(),
+            ),
+        )
+
+        if sched_backbone_config is not None and sched_downstream_config is not None:
+
+            scheduler_backbone = WarmupPolicy(
+                optimizer=optim_backbone,
+                max_steps=None,
+                min_lr=sched_backbone_config.min_lr,
+                warmup_steps=sched_backbone_config.warmup_steps,
+            )  # Use warmup to delay start
+            sch1_dict = {
+                'scheduler': scheduler_backbone,
+                'interval': 'step',
+            }
+
+            scheduler_downstream = WarmupPolicy(
+                optimizer=optim_downstream,
+                max_steps=None,
+                min_lr=sched_downstream_config.min_lr,
+                warmup_steps=sched_downstream_config.warmup_steps,
+            )
+            sch2_dict = {
+                'scheduler': scheduler_downstream,
+                'interval': 'step',
+            }
+
+            return [optim_backbone, optim_downstream], [sch1_dict, sch2_dict]
+        else:
+            return [optim_backbone, optim_downstream]
+
+    def get_mel_spectrogram(self, audio, audio_len):
```

Comment:
Add docstring, though also--should this be a generic function in helpers rather than in this class?

Reply:
removed this. Was being used for speaker loss setup for fastpitch, no longer needed.